src/transformers/modeling_bart.py

```diff
@@ -0,0 +1,970 @@
+# coding=utf-8
+# Copyright 2018 The Google AI Language Team Authors and The HuggingFace Inc. team.
+# Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PyTorch BART model, ported from the fairseq repo."""
+
+import logging
+import random
+from collections import namedtuple
+from typing import Dict, List, Optional, Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor, nn
+
+from .configuration_bart import BartConfig
+from .file_utils import add_start_docstrings
+from .modeling_utils import PreTrainedModel
+
+
+logger = logging.getLogger(__name__)
+
+
+BART_PRETRAINED_MODEL_ARCHIVE_MAP = {
+    "bart-large": "https://s3.amazonaws.com/models.huggingface.co/bert/bart-large-pytorch_model.bin",
+    "bart-large-mnli": "https://s3.amazonaws.com/models.huggingface.co/bert/bart-large-mnli-pytorch_model.bin",
+}
+
+
+BART_START_DOCSTRING = r"""  TODO(SS): FIXME"""
+
+
+class PretrainedBartModel(PreTrainedModel):
+    config_class = BartConfig
+    base_model_prefix = "model"
+    pretrained_model_archive_map = BART_PRETRAINED_MODEL_ARCHIVE_MAP
+
+    def _init_weights(self, module):
+        std = self.init_std  # used by init_params
+
+        # called init_bert_params in fairseq
+        if isinstance(module, nn.Linear):
```

Comment:
Note that we usually also initialize the `LayerNorm` for completeness even though they are now initialized by default to the standard initialization in PyTorch so you don't really need to perpetuate this practice.